pkg/log/log.go

```
@@ -0,0 +1,278 @@
+	for _, dest := range config.destinations {
+		switch dest.Type {
+		case Stdout:
+			writers = append(writers, os.Stdout)
+		case File:
+			filePath := dest.Config["path"]
+			dir := filepath.Dir(filePath)
+			if err := os.MkdirAll(dir, os.ModePerm); err != nil {
+				return newLogger, fmt.Errorf("failed to create log directory: %v", err)
+			}
+
+			fmt.Printf("writing test log to file: %v\n", config)
+			lumberjackLogger := &lumberjack.Logger{
+				Filename:   filePath,
+				MaxSize:    500, // Default size in MB if not overridden
+				MaxBackups: 15,  // Number of backups
+				MaxAge:     30,  // Days to retain
+				Compress:   false,
+			}
+			absPath, err := filepath.Abs(filePath)
+			if err != nil {
+				return newLogger, fmt.Errorf("failed to get absolute path: %v", err)
+			}
+			fmt.Printf("Attempting to write logs to: %s\n", absPath)
+			lumberjackLogger.Filename = absPath
+
+			setConfigValue := func(key string, target *int) {
+				if valStr, ok := dest.Config[key]; ok {
+					if val, err := strconv.Atoi(valStr); err == nil {
+						*target = val
+					}
+				}
+			}
+			setConfigValue("maxSize", &lumberjackLogger.MaxSize)
+			setConfigValue("maxBackups", &lumberjackLogger.MaxBackups)
+			setConfigValue("maxAge", &lumberjackLogger.MaxAge)
+			if compress, ok := dest.Config["compress"]; ok {
+				lumberjackLogger.Compress = compress == "true"
+			}
+			writers = append(writers, lumberjackLogger)
+		}
+	}
+	multiwriter := io.MultiWriter(writers...)
+	defer func() {
+		if closer, ok := multiwriter.(io.Closer); ok {
+			closer.Close()
+		}
+	}()
+	newLogger = zerolog.New(multiwriter).
+		Level(logLevels[config.level]).
+		With().
+		Timestamp().
+		Caller().
+		Logger()
+
+	cfg = config
+	return newLogger, nil
+}
+func InitLogger(c Config) error {
+	var initErr error
+	once.Do(func() {
+		if err := c.Validate(); err != nil {
+			return
+		}
+
+		logger, initErr = getLogger(c)
+		if initErr != nil {
+			return
+		}
+	})
+	return initErr
```

If config validation fails, error will never be reported as err is never returned 
err can be replaced initErr and can be handled properly

---

done